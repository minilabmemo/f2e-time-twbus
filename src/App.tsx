
import './style/notosanstc.css';
import './style/notosans.css';
import './style/style.css';
import { Header } from './components/layout/header';
import PetFinder from './pages/CityBusHome';
import { Footer } from './components/layout/Footer';
import { HashRouter, Route, Routes } from 'react-router-dom';
import { URI_NEARBY_DEFAULT, URI_PET_FIND_PREFIX, URI_SEARCH, URI_STOPS } from './utils/const';
import { Wait } from './components/Wait';
import { BusRouteSearch } from './pages/BusRouteSearch';
import { BusRouteStops } from './pages/BusRouteStops';
import { BusRouteStopsNearBy } from './pages/BusRouteStopsNearBy';


const BaseLayout = ({ children }: { children: React.ReactNode }) => (
  <div className="base">
    <div className="container">
      <Header></Header>
      {children}
      <Footer></Footer>
    </div>
  </div>
);

function App() {
  return (
    <HashRouter>
      <Routes>
        <Route path="/" element={<BaseLayout><PetFinder /></BaseLayout>} />
        <Route path={`${URI_NEARBY_DEFAULT}`} element={<BaseLayout><BusRouteStopsNearBy /></BaseLayout>} />
        <Route path={`${URI_SEARCH}`} element={<BaseLayout><BusRouteSearch /></BaseLayout>} />
        <Route path={`${URI_STOPS}`} element={<BaseLayout><BusRouteStops /></BaseLayout>} />
      </Routes>
    </HashRouter>
  );
}

export default App;
