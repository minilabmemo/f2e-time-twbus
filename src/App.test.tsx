import React from 'react';

import { render, screen } from '@testing-library/react';
import App from './App';

// test('render app component', () => {
//   render(<App />);
//   const headerElement = screen.getByTestId('header-component');
//   expect(headerElement).toBeInTheDocument();
// });

test('render app toMatchSnapshot', () => {
  const view = render(<App />);
  expect(view).toMatchSnapshot();
});

test('renders the title with the correct text', () => {
  render(<App></App>);
  const titleElement = screen.getByText('台 灣 公 車 動 態');
  expect(titleElement).toBeInTheDocument();
});
