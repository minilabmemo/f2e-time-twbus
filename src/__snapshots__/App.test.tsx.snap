// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app toMatchSnapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="base"
      >
        <div
          class="container"
        >
          <div
            class="header"
          >
            <div
              class="brand"
            >
              <a
                aria-current="page"
                class="home-link active"
                href="#/"
              >
                <img
                  alt="logo"
                  height="42px"
                  src="logo.svg"
                  width="162px"
                />
              </a>
            </div>
            <div
              class="nav"
            >
              <a
                class="nav-link"
                href="#/nearby"
              >
                附近站牌
              </a>
              <a
                class="nav-link"
                href="#/search/zh/cities"
              >
                站點查詢
              </a>
              <a
                class="nav-link"
                href="#/save/zh"
              >
                我的收藏 
              </a>
            </div>
            <div
              class="lang"
            >
              <img
                alt="icon_save"
                height="20px"
                src="icon_language.svg"
                width="20px"
              />
              <div>
                <span
                  class="active"
                >
                   中文
                </span>
                ｜
                <span>
                  英文
                </span>
              </div>
            </div>
            <div
              class="phone-options"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-bars "
                data-icon="bars"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
          <section
            class="home-main"
          >
            <div
              class="photo_wrap"
            >
              <div
                class="main-text"
              >
                <div>
                  taiwan bus+
                </div>
                <div>
                  台 灣 公 車 動 態
                </div>
                <div>
                  時 刻 查 詢 系 統
                </div>
              </div>
            </div>
          </section>
          <section
            class="home-cities"
          >
            <a
              class="city-card city-color-black city-link"
              href="#/search/zh/Taipei"
            >
              <div
                class="city-icon"
              >
                 
                <svg
                  fill="#283C43"
                  viewBox="0 0 51 51"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                  />
                  <path
                    d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                  />
                  <path
                    d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                  />
                  <path
                    d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                  />
                  <path
                    d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                  />
                </svg>
              </div>
              <div
                class="city-name"
              >
                台北市 
              </div>
              <div
                class="city-name-en"
              >
                 Taipei 
              </div>
            </a>
            <a
              class="city-card city-color-blue city-link"
              href="#/search/zh/Taoyuan"
            >
              <div
                class="city-icon"
              >
                 
                <svg
                  fill="#3591C5"
                  viewBox="0 0 51 51"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                  />
                  <path
                    d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                  />
                  <path
                    d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                  />
                  <path
                    d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                  />
                  <path
                    d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                  />
                </svg>
              </div>
              <div
                class="city-name"
              >
                桃園市
              </div>
              <div
                class="city-name-en"
              >
                Taoyuan
              </div>
            </a>
            <a
              class="city-card city-color-pink city-link"
              href="#/search/zh/Taichung"
            >
              <div
                class="city-icon"
              >
                 
                <svg
                  fill="#D08181"
                  viewBox="0 0 51 51"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                  />
                  <path
                    d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                  />
                  <path
                    d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                  />
                  <path
                    d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                  />
                  <path
                    d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                  />
                </svg>
              </div>
              <div
                class="city-name"
              >
                台中市
              </div>
              <div
                class="city-name-en"
              >
                Taichung
              </div>
            </a>
            <a
              class="city-card city-color-cyan city-link"
              href="#/search/zh/Tainan"
            >
              <div
                class="city-icon"
              >
                 
                <svg
                  fill="#7FC0C5"
                  viewBox="0 0 51 51"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                  />
                  <path
                    d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                  />
                  <path
                    d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                  />
                  <path
                    d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                  />
                  <path
                    d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                  />
                </svg>
              </div>
              <div
                class="city-name"
              >
                台南市
              </div>
              <div
                class="city-name-en"
              >
                Tainan
              </div>
            </a>
            <a
              class="city-card city-color-blue-light city-link"
              href="#/search/zh/Kaohsiung"
            >
              <div
                class="city-icon"
              >
                 
                <svg
                  fill="#5E9BAE"
                  viewBox="0 0 51 51"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                  />
                  <path
                    d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                  />
                  <path
                    d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                  />
                  <path
                    d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                  />
                  <path
                    d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                  />
                </svg>
              </div>
              <div
                class="city-name"
              >
                高雄市
              </div>
              <div
                class="city-name-en"
              >
                 Kaohsiung
              </div>
            </a>
            <a
              class="city-card city-color-blue-dark city-link"
              href="#/search/zh/cities"
            >
              <div
                class="city-icon"
              >
                 
                <svg
                  fill="#52797C"
                  viewBox="0 0 51 51"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                  />
                  <path
                    d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                  />
                  <path
                    d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                  />
                  <path
                    d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                  />
                  <path
                    d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                  />
                </svg>
              </div>
              <div
                class="city-name"
              >
                其他地區
              </div>
              <div
                class="city-name-en"
              >
                Other City
              </div>
            </a>
          </section>
          <div
            class="footer"
          >
            <span
              class="footer-right"
            >
              Copyright © 2023 台灣公車 e 點通 All rights reserved.
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="base"
    >
      <div
        class="container"
      >
        <div
          class="header"
        >
          <div
            class="brand"
          >
            <a
              aria-current="page"
              class="home-link active"
              href="#/"
            >
              <img
                alt="logo"
                height="42px"
                src="logo.svg"
                width="162px"
              />
            </a>
          </div>
          <div
            class="nav"
          >
            <a
              class="nav-link"
              href="#/nearby"
            >
              附近站牌
            </a>
            <a
              class="nav-link"
              href="#/search/zh/cities"
            >
              站點查詢
            </a>
            <a
              class="nav-link"
              href="#/save/zh"
            >
              我的收藏 
            </a>
          </div>
          <div
            class="lang"
          >
            <img
              alt="icon_save"
              height="20px"
              src="icon_language.svg"
              width="20px"
            />
            <div>
              <span
                class="active"
              >
                 中文
              </span>
              ｜
              <span>
                英文
              </span>
            </div>
          </div>
          <div
            class="phone-options"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-bars "
              data-icon="bars"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <section
          class="home-main"
        >
          <div
            class="photo_wrap"
          >
            <div
              class="main-text"
            >
              <div>
                taiwan bus+
              </div>
              <div>
                台 灣 公 車 動 態
              </div>
              <div>
                時 刻 查 詢 系 統
              </div>
            </div>
          </div>
        </section>
        <section
          class="home-cities"
        >
          <a
            class="city-card city-color-black city-link"
            href="#/search/zh/Taipei"
          >
            <div
              class="city-icon"
            >
               
              <svg
                fill="#283C43"
                viewBox="0 0 51 51"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                />
                <path
                  d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                />
                <path
                  d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                />
                <path
                  d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                />
                <path
                  d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                />
              </svg>
            </div>
            <div
              class="city-name"
            >
              台北市 
            </div>
            <div
              class="city-name-en"
            >
               Taipei 
            </div>
          </a>
          <a
            class="city-card city-color-blue city-link"
            href="#/search/zh/Taoyuan"
          >
            <div
              class="city-icon"
            >
               
              <svg
                fill="#3591C5"
                viewBox="0 0 51 51"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                />
                <path
                  d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                />
                <path
                  d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                />
                <path
                  d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                />
                <path
                  d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                />
              </svg>
            </div>
            <div
              class="city-name"
            >
              桃園市
            </div>
            <div
              class="city-name-en"
            >
              Taoyuan
            </div>
          </a>
          <a
            class="city-card city-color-pink city-link"
            href="#/search/zh/Taichung"
          >
            <div
              class="city-icon"
            >
               
              <svg
                fill="#D08181"
                viewBox="0 0 51 51"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                />
                <path
                  d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                />
                <path
                  d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                />
                <path
                  d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                />
                <path
                  d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                />
              </svg>
            </div>
            <div
              class="city-name"
            >
              台中市
            </div>
            <div
              class="city-name-en"
            >
              Taichung
            </div>
          </a>
          <a
            class="city-card city-color-cyan city-link"
            href="#/search/zh/Tainan"
          >
            <div
              class="city-icon"
            >
               
              <svg
                fill="#7FC0C5"
                viewBox="0 0 51 51"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                />
                <path
                  d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                />
                <path
                  d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                />
                <path
                  d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                />
                <path
                  d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                />
              </svg>
            </div>
            <div
              class="city-name"
            >
              台南市
            </div>
            <div
              class="city-name-en"
            >
              Tainan
            </div>
          </a>
          <a
            class="city-card city-color-blue-light city-link"
            href="#/search/zh/Kaohsiung"
          >
            <div
              class="city-icon"
            >
               
              <svg
                fill="#5E9BAE"
                viewBox="0 0 51 51"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                />
                <path
                  d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                />
                <path
                  d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                />
                <path
                  d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                />
                <path
                  d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                />
              </svg>
            </div>
            <div
              class="city-name"
            >
              高雄市
            </div>
            <div
              class="city-name-en"
            >
               Kaohsiung
            </div>
          </a>
          <a
            class="city-card city-color-blue-dark city-link"
            href="#/search/zh/cities"
          >
            <div
              class="city-icon"
            >
               
              <svg
                fill="#52797C"
                viewBox="0 0 51 51"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M43.0312 17.5312H46.2188V23.9062H43.0312V17.5312Z"
                />
                <path
                  d="M4.78125 17.5312H7.96875V23.9062H4.78125V17.5312Z"
                />
                <path
                  d="M31.875 31.875H35.0625V35.0625H31.875V31.875Z"
                />
                <path
                  d="M15.9375 31.875H19.125V35.0625H15.9375V31.875Z"
                />
                <path
                  d="M33.4688 6.375H17.5312C15.4186 6.37753 13.3932 7.21791 11.8993 8.71179C10.4054 10.2057 9.56503 12.2311 9.5625 14.3438V36.6562C9.56334 37.5014 9.89944 38.3116 10.497 38.9092C11.0946 39.5068 11.9049 39.8429 12.75 39.8438V44.625H15.9375V39.8438H35.0625V44.625H38.25V39.8438C39.095 39.8425 39.905 39.5063 40.5025 38.9088C41.1 38.3113 41.4362 37.5012 41.4375 36.6562V14.3438C41.435 12.2311 40.5946 10.2057 39.1007 8.71179C37.6068 7.21791 35.5814 6.37753 33.4688 6.375ZM38.25 15.9375V25.5H12.75V15.9375H38.25ZM17.5312 9.5625H33.4688C34.4544 9.56543 35.4149 9.8734 36.2185 10.4441C37.0221 11.0149 37.6293 11.8203 37.9568 12.75H13.0433C13.3707 11.8203 13.9779 11.0149 14.7815 10.4441C15.5851 9.8734 16.5456 9.56543 17.5312 9.5625ZM12.75 36.6562V28.6875H38.2516L38.2532 36.6562H12.75Z"
                />
              </svg>
            </div>
            <div
              class="city-name"
            >
              其他地區
            </div>
            <div
              class="city-name-en"
            >
              Other City
            </div>
          </a>
        </section>
        <div
          class="footer"
        >
          <span
            class="footer-right"
          >
            Copyright © 2023 台灣公車 e 點通 All rights reserved.
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
